databroker/queries.py

```diff
@@ -341,6 +382,41 @@ def scan_id(query, catalog):
     return results
 
 
+def scan_id_range(query, catalog):
+    mongo_results = catalog.apply_mongo_query({"scan_id": {"$gt": query.start_id, "$lt": query.end_id}})
```

Comment:
I think users will expect an inclusive lower bound. `ScanIDRange(5, 10)` should include `5` for sure. What to do with the upper bound is less clear. We could follow the behavior of `range` user a half-open interval, or we could make it fully closed.

@michael-sutherland What would you expect here?

Reply:
I'd expect the same behavior as range, half-open interval.